pkg/utils/env/env.go

```diff
@@ -33,6 +33,20 @@ func WithDefaultInt(key string, def int) int {
 	return i
 }
 
+// WithDefaultFloat64 returns the float64 value of the supplied environment variable or, if not present,
+// the supplied default value. If the float64 conversion fails, returns the default
+func WithDefaultFloat64(key string, def float64) float64 {
```

Comment:
Is there a generics way to do this?

Reply:
I don't think so.  The float64 func uses `strconv.ParseFloat`, the Bool one uses strconv.ParseBool and the Int one uses `strconv.Atoi`.  I think the only way you could do it is to use a generic return type and accept a default interface{} as the parameter so that you could do a type switch:

```
func WithDefault[V float64 | int | string | bool](key string, def V) V {
    var err error
    var p V
    val, ok := os.LookupEnv(key)
    if !ok {
        return def
    }
    switch def.(type) {
        case float64:
           p, err = strconv.ParseFloat(val, 64)
        case int:
           p, err = strconv.Atoi(val)
        case bool:
           p, err = strconv.ParseBool(val)
       default:
           panic("not supported")
    }
    if err != nil {
        return def.(V)
    }
    return p
}
```

I think they're better as separate funcs :)